### ISSUE TITLE
Slow query (or manual delay) causes error and result set to be blank 

### ISSUE THREAD

**Issue by wilcoxjay** [microsoft/vscode-mssql#17819]
Type: <b>Bug</b>

I am connected to a SQL Server instance. Running any simple query is fine, but when the queries start taking more than a few hundred ms, the result sets are blank.

For example

```sql
SELECT 'hello' as msg;
```

works fine, but 

```sql
WAITFOR DELAY '00:00:00.5';  -- ie, wait 500ms
SELECT 'hello' AS msg;
```

gives a blank result set. Interestingly, I can copy the data out of the result cell and paste it into a file and the data is present. It's just not rendering on the screen.

Here is a recording of me demonstrating the issue. I run the query without delay and it is fine. I run the query with delay and get an error and the result set is blank.

https://github.com/microsoft/vscode-mssql/assets/229924/de69d224-f1d4-4a37-9071-16c57b83f317

Here is a screenshot of the final state of the screen from that recording.
![Screenshot 2023-09-20 at 14 22 45](https://github.com/microsoft/vscode-mssql/assets/229924/88c9f8e0-3fea-45de-bc8b-7bcbbc218670)

And here is the contents of the console log in plain text: [vscode-app-1695245539102.log](https://github.com/microsoft/vscode-mssql/files/12677693/vscode-app-1695245539102.log)

Extension version: 1.20.1
VS Code version: Code 1.82.2 (abd2f3db4bdb28f9e95536dfa84d8479f1eb312d, 2023-09-14T06:00:27.244Z)
OS version: Darwin arm64 22.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (0.09GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by cheenamalhotra** [comment]
@wilcoxjay 

I tried the same, and can't reproduce the issue. I tried with 1.21.0 (released today) but I tested Windows. The behavior should be same as far as I can tell. Can you give the latest version a try and confirm?

**Comment by wilcoxjay** [comment]
Hmm, after updating to 1.21.0 and reloading VS Code, I can't reproduce. But then I downgraded to 1.20.1 again and reloaded and I can't reproduce it any more there either! So I don't think it was fixed by 1.21.0, but something fishier is going on.

I think we can close this issue until I can reproduce it again.

**Comment by cheenamalhotra** [comment]
Sure, let us know if you face it again or find a way to reproduce it.
